src/Compilers/Core/Portable/MetadataReader/MetadataDecoder.cs

```diff
@@ -1095,15 +1109,26 @@ private void DecodeParameterOrThrow(ref BlobReader signatureReader, /*out*/ ref
         {
             bool refersToNoPiaLocalType;
 
+            var isConstFound = false;
             SignatureTypeCode typeCode;
-            info.CustomModifiers = DecodeModifiersOrThrow(ref signatureReader, out typeCode);
+            info.CustomModifiers = DecodeModifiersOrThrow(ref signatureReader, out typeCode, acceptRequiredModifier: type =>
+            {
+                var isConst = IsConstModifierType(type);
+                isConstFound |= isConst;
+                return isConst;
+            });
 
             if (typeCode == SignatureTypeCode.ByReference)
             {
                 info.IsByRef = true;
-                info.RefCustomModifiers = info.CustomModifiers; 
+                info.RefCustomModifiers = info.CustomModifiers;
                 info.CustomModifiers = DecodeModifiersOrThrow(ref signatureReader, out typeCode);
             }
+            else if (isConstFound)
+            {
+                // This cannot be placed on CustomModifiers, just RefCustomModifiers
+                throw new UnsupportedSignatureContent();
+            }
```

Comment:
Will remove the need for capturing, but we still need that `isConstFound` value since we allow optional modifiers, just not required ones.

---
In reply to: [117907098](https://github.com/dotnet/roslyn/pull/19658#discussion_r117907098) [](ancestors = 117907098)